chore: #0 프로젝트 기본환경셋팅 입니다

paulus-seo opened the pull request:
## Django 프로젝트 환경셋팅

#### _Django 언어 및 시간 설정 변경을 실시했습니다._
- Django Tutorial을 보면서 간단하게 언어 및 시간 설정을 한국으로 변경해봤습니다.

#### _Django와 MySQL 연동을 실시했습니다._
- Django는 기본적으로 sqlite3를 지원했고, 프로젝트 간 MySQL을 사용하기로 결정했으므로 해당 환경셋팅 변경을 실시했습니다.

#### _settings.py 설정파일 분리를 실시했습니다._
- 보안과 배포 상의 이유로 설정파일 분리를 실시했습니다.
- settings.py에 있는 SECRET_KEY와 DATABASES 는 개인정보이므로 원격저장소에 올라가면 보안 상의 위험이 발생할 수 있습니다. 따라서 JSON파일에 해당 정보를 넣고, gitignore에 해당 JSON파일을 추가해서 원격저장소에는 올라가지 않고 개인만이 보관하도록 만들었습니다.
- 개발환경과 배포환경의 설정은 서로 다를 수 있으므로, 공통되는 settings패키지를 만들어 공통된 환경설정은 base.py, 개발환경에 관한 설정은 development.py, 배포환경에 관한 설정은 production.py에 올리도록 파일을 분리했습니다.

#### _의존성 관련 파일 추가 및 Code Formatter 설치를 실시했습니다._
- 내 개발환경에서 이용된 모듈의 버전에 맞도록 자동설치하는 의존성 관련 파일을 추가했습니다.
- pip freeze > requirements.txt 명령어를 이용해서 쉽게 관련 파일을 추가할 수 있었습니다.
- autopep8이라는 Code Formatter를 설치해서 실행시켜봤습니다. (자동으로 코딩 스타일 수정도 해줘서 잘 이용하면 개발에 도움이 많이 될 것 같다고 느꼈습니다.)

paulus-seo committed:
chore: #0 프로젝트 기본환경셋팅 입니다

Django 프로젝트 환경셋팅 실시했습니다.
settings.py 에서는 다음 내용을 변경했습니다.
1) 영어 -> 한국어 언어코드 변경.
2) 영국 -> 한국 시간대 변경.
3) static파일 경로지정.

paulus-seo committed:
chore: #0 Django와 MySQL 연동 기본환경셋팅 입니다

Django와 MySQL 연동을 위해서 settings.py 파일을 수정했습니다.
Django에서 기본으로 사용하는 sqlite3.db 대신에 로컬에 설치된 MySQL과 연동하도록 변수를 셋팅했습니다.

paulus-seo committed:
chore: #0 프로젝트 내 작업공간을 생성했습니다

프로젝트 디렉토리인 tcpsite 내부에 앱 디렉토리(작업공간) tcp를 생성했습니다.
기본적인 프로젝트 셋팅을 마쳤으며, 이후부터 1주차 개발에 들어갈 예정입니다.

paulus-seo committed:
chore: #0 Django Rest Framework를 추가했습니다

Django에서 Restful API 서버를 쉽게 구축하도록 도와주는 오픈 소스 라이브러리 라고 합니다.
settings.py 에 해당 Framework의 경로 및 설정을 추가했습니다.

paulus-seo committed:
chore: #0 GitHub에 추가된 불필요한 파일을 삭제 및 조정했습니다

.idea 에 관련된 내용을 develop 브랜치에서 삭제했습니다.
gitignore에 .idea 관련 디렉토리를 무시하도록 설정했습니다.

paulus-seo committed:
chore: #0 settings.py 설정파일을 분리했습니다

SECRET_KEY와 DATABASES 정보 등 보안에 위험성 있는 파일을 JSON으로 분리했습니다.
해당 파일은 .gitignore에 추가해서 원격저장소에 올라가지 못하도록 설정하여 보안성을 높였습니다.
util.py는 내가 가지고 있는 JSON파일을 파싱해 각각 SECRET_KEY와 DATABASES에 넣어주는 기능을 가지고 있습니다.

settings.py 설정파일을 DEV와 PROD로 각각 분리했습니다.
개발환경과 배포환경에서 필요한 페이지 및 설정이 각각 다를 수 있으므로 이를 분리하여 배포 시 변경 수정을 최소화했습니다.

paulus-seo committed:
fix: #0 cmd에서 DJANGO_SETTINGS_MODULE 환경변수를 수정하지 않으면 실행이 안되는 문제를 수정했습니다

DJANGO_SETTINGS_MODULE 환경변수를 development 또는 production으로 변경하지 않으면 실행이 되지 않는 문제를 수정했습니다.
settings.py 파일을 패키지로 만들면서 manage.py 의 환경변수 Default 값을 변경하지 않아 발생한 문제였습니다.
Default값은 개발환경으로 맞춰놨습니다.

lallaheeee reviewed (approved):
LGTM ⚡️
덕분에 파이썬도 찾아보게되네유~~환경세팅하느라 고생하셨어유  화이팅합시다 !!
근데 의존성 관련 파일은 보이지 않는데 파이썬 의존성은 어떻게 관리하나유? 

검색하니까 다음과 같은 글을 찾을 수 있었는데
https://velog.io/@ground4ekd/python-requirements

린트랑 포맷터 설치도 해봅시다 ~~ㅎㅎ 

https://github.com/gothinkster/django-realworld-example-app
이건 장고 realworld example인데유 
장고 처음이시니까 참고하면 많이 도움되실거같아유

paulus-seo committed:
chore: #0 의존성 관련 파일 생성 및 Code Formatter 설치

pip freeze > requirements.txt 명령어를 이용해 requirements.txt 라는 의존성 관련 파일을 생성했습니다.
autopep8 이라는 Code Formatter를 설치하고 autopep8 -i manage.py 라는 명령어로 manage.py의 코드를 검사 후 변경했습니다.
(위 변경사항 중 tcpsite/manage.py 변경은 autopep8에 의한 변경사항입니다.)
